src/protocol/codecs/topic.rs

```diff
@@ -28,9 +30,17 @@ impl MsgOfInterest {
             return Err(invalid_data!("expected 'tags' topic"));
         }
 
-        Ok(Self {
-            tags: tag_topic.value,
-        })
+        let tags = tag_topic.value.split(',');
+        let expected_len = tags.clone().count();
```

Comment:
Do we need this `clone` here?

Reply:
We do need to copy it because `count()` consumes the iterator

Reply:
IDK how large `tags` usually is, or how performance critical this path is, but how about:
`let expected_len = 1 + tag_topic.value.chars().filter(|c| *c == ',').count();`
On my machine that's like 30% quicker.

Reply:
Aha, good to know!
The tags are pretty short here. Here's an example of what we receive from the node:
`UE,NP,SP,TS,MI,TX,UC,MS,pi,AV,pj,PP,VB`

Do you think we should apply your suggestion here?
The solution you find more readable is the solution I vote for 🤔

Reply:
Actually, I think my suggestion is less readable, but 30% performance hit is not something I give up easily. IDK how hot this path is, so your call :)

Reply:
We receive this message only once after the handshake, so the path is quite a cold one 🧊